app/lib/bk/compat/parsers/circleci/orbs.rb

```diff
@@ -64,6 +61,54 @@ def install_commands(version, install_dir)
           end
         end
 
+        def translate_docker_build(config)
+          [
+            ("# Attaching workspace at #{config['attach-at']}" if config['attach-at']),
+            ('export DOCKER_BUILDKIT=1' if config['use-buildkit']),
+            lint_dockerfile(config),
+            *pull_cache_images(config),
+            generate_build_command(config)
+          ].compact
+        end
+
+        def lint_dockerfile(config)
+          return unless config['lint-dockerfile']
+
+          "# Linting Dockerfile#{' (treating warnings as errors)' if config['treat-warnings-as-errors']}"
+        end
+
+        def pull_cache_images(config)
+          config['cache_from']&.split(',')&.map { |image| "docker pull #{image.strip} || true" } || []
```

Comment:
the nil-safe operators can be avoided with appropriate defaults:

```suggestion
            config.fetch('cache-from', '').split.map { |image| 
              "docker pull #{image.strip} || true"
            } 
```